include/enrico/surrogate_heat_driver.h

```diff
@@ -73,6 +219,32 @@ class SurrogateHeatDriver : public HeatFluidsDriver {
   //! Create internal arrays used for heat equation solver
   void generate_arrays();
 
+  //! Channel index in terms of row, column index
+  int channel_index(int row, int col) const { return row * (n_pins_x_ + 1) + col; }
+
+  //! Rod power at a given node in a given pin, computed by integrating the heat source
+  //! (assumed constant in each ring) over the pin.
+  //! \param pin   pin index
+  //! \param axial axial index
+  double rod_axial_node_power(const int pin, const int axial) const;
+
+  //! Diagnostic function to assess whether the mass is conserved by the subchannel
+  //! solver by comparing the mass flowrate in each axial plane (at cell-centered
+  //! positions) to the specified inlet mass flowrate.
+  //! \param rho density in a cell-centered basis
+  //! \param u   axial velocity in a face-centered basis
+  bool mass_conservation(const xt::xtensor<double, 2>& rho, const xt::xtensor<double, 2>& u) const;
```

Comment:
`is_mass_conserved` would be more descriptive since this returns `bool` (same thing with energy below)

Reply:
Yeah, I agree on that